internal/provider/slo_resource.go

```diff
@@ -0,0 +1,460 @@
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.Map{
+					modifiers.EquivalentTags(),
+				},
+				Validators: []validator.Map{
+					mapvalidator.SizeAtMost(client.MaxTagsPerResource),
+					mapvalidator.KeysAre(
+						stringvalidator.RegexMatches(client.TagKeyValidationRegex, "must only contain lowercase letters, and be 1-32 characters long"),
+					),
+					mapvalidator.ValueStringsAre(
+						stringvalidator.RegexMatches(client.TagValueValidationRegex, "must begin with a lowercase letter, be between 1-32 characters long, and only contain lowercase alphanumeric characters, -, or /"),
+					),
+				},
+			},
+		},
+	}
+}
+
+func (r *sloResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	dataset, id, found := strings.Cut(req.ID, "/")
+
+	// if dataset separator not found, we will assume its the bare id
+	// if thats the case, we need to reassign values since strings.Cut would return (id, "", false)
+	dsValue := types.StringNull()
+	idValue := types.StringValue(id)
+	if !found {
+		idValue = types.StringValue(dataset)
+	} else {
+		dsValue = types.StringValue(dataset)
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &models.SLOResourceModel{
+		ID:       idValue,
+		Dataset:  dsValue,
+		Datasets: types.SetNull(types.StringType),
+		Tags:     types.MapNull(types.StringType),
+	})...)
+}
+
+func (r *sloResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan models.SLOResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	dataset := helper.GetDatasetOrAll(plan.Dataset)
+	expandedSLO, err := expandSLO(ctx, plan)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error expanding SLO for creation",
+			fmt.Sprintf("Could not expand SLO: %s", err),
+		)
+		return
+	}
+
+	slo, err := r.client.SLOs.Create(ctx, dataset.ValueString(), expandedSLO)
+	if helper.AddDiagnosticOnError(&resp.Diagnostics, "Creating Honeycomb SLO", err) {
+		return
+	}
+
+	var state models.SLOResourceModel
+	state.ID = types.StringValue(slo.ID)
+	state.Name = types.StringValue(slo.Name)
+	state.Description = types.StringValue(slo.Description)
+	state.SLI = types.StringValue(slo.SLI.Alias)
+	state.TargetPercentage = types.Float64Value(helper.PPMToFloat(slo.TargetPerMillion))
+	state.TimePeriod = types.Int64Value(int64(slo.TimePeriodDays))
+
+	if !plan.Dataset.IsNull() {
+		state.Dataset = plan.Dataset
+	}
+
+	// Convert the slice of strings to a types.Set
+	if len(slo.DatasetSlugs) > 0 {
+		ds := slo.DatasetSlugs
+		if ds == nil {
+			ds = []string{}
+		}
+		datasetsSet, diags := types.SetValueFrom(ctx, types.StringType, ds)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		state.Datasets = datasetsSet
+	} else {
+		// Create an empty set if there are no dataset slugs
+		state.Datasets = types.SetNull(types.StringType)
+	}
```

Comment:
this is repeated often: might be nice to pull this into a function like `flattenDatasetSlugs`